docker/docker_image_dest.go

```diff
@@ -120,15 +151,25 @@ func (d *dockerImageDestination) PutBlob(stream io.Reader, inputInfo types.BlobI
 			return types.BlobInfo{Digest: inputInfo.Digest, Size: size}, nil
 		}
 	}
+	mountQuery := prepareMountQuery(d.mounts, inputInfo.Digest.String())
 
 	// FIXME? Chunked upload, progress reporting, etc.
 	uploadPath := fmt.Sprintf(blobUploadPath, reference.Path(d.ref.ref))
+	if mountQuery != "" {
```

Comment:
Nit: it would be cleaner to have a single chunk dealing with mounts, and maybe to have all of it conditional on `len(d.mounts != 0)`, instead of treating the `""` value a bit magically.

Reply:
This was simplified a bit by dropping support for multiple mounts.